new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAddCluster.java

```diff
@@ -0,0 +1,998 @@
+        String.format("createSecret failed for %s", REPO_SECRET_NAME));
+    assertTrue(secretCreated, String.format("createSecret failed while creating secret %s in namespace",
+                  REPO_SECRET_NAME, opNamespace));
+
+    // map with secret
+    secretNameMap = new HashMap<String, Object>();
+    secretNameMap.put("name", REPO_SECRET_NAME);
+    // helm install parameters
+    opHelmParams = new HelmParams()
+        .releaseName(OPERATOR_RELEASE_NAME)
+        .namespace(opNamespace)
+        .chartDir(OPERATOR_CHART_DIR);
+
+    // Operator chart values to override
+    OperatorParams opParams =
+        new OperatorParams()
+            .helmParams(opHelmParams)
+            .image(operatorImage)
+            .imagePullSecrets(secretNameMap)
+            .domainNamespaces(Arrays.asList(domainNamespace))
+            .serviceAccount(serviceAccountName);
+
+    // install Operator
+    logger.info("Installing Operator in namespace {0}", opNamespace);
+    assertTrue(installOperator(opParams),
+        String.format("Operator install failed in namespace %s", opNamespace));
+    logger.info("Operator installed in namespace {0}", opNamespace);
+
+    // list helm releases matching Operator release name in operator namespace
+    logger.info("Checking Operator release {0} status in namespace {1}",
+        OPERATOR_RELEASE_NAME, opNamespace);
+    assertTrue(isHelmReleaseDeployed(OPERATOR_RELEASE_NAME, opNamespace),
+        String.format("Operator release %s is not in deployed status in namespace %s",
+            OPERATOR_RELEASE_NAME, opNamespace));
+    logger.info("Operator release {0} status is deployed in namespace {1}",
+        OPERATOR_RELEASE_NAME, opNamespace);
+
+    // check operator is running
+    logger.info("Check Operator pod is running in namespace {0}", opNamespace);
+    withStandardRetryPolicy
+        .conditionEvaluationListener(
+            condition -> logger.info("Waiting for operator to be running in namespace {0} "
+                    + "(elapsed time {1}ms, remaining time {2}ms)",
+                opNamespace,
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        .until(operatorIsReady(opNamespace));
+  }
+
+  /**
+   * Start a WebLogic domain without any pre-defined ConfigMap.
+   * Create a ConfigMap with a sparse model file to add a dynamic cluster.
+   * Patch the domain resource with the ConfigMap.
+   * Patch the domain resource with the spec/replicas set to 1
+   * Update the restart version of the domain resource to 2
+   * Verify rolling restart of the domain by comparing PodCreationTimestamp before and after rolling restart.
+   * Verify servers from the newly added cluster are in running state.
+   */
+  @Test
+  @Order(1)
+  @DisplayName("Add a dynamic cluster to model in image domain")
+  @Slow
+  @MustNotRunInParallel
+  public void testAddMiiDynamicCluster() {
```

Comment:
This is what I think you should do for these tests, 

- create the domain resource in beforeAll method
- verify the domain and cluster(if any existing) is running in beforeEach method
- Tests should be just adding new cluster/patch/verify rolling restart etc.. no need for ordering, tests can run in an order...lot of code can be moved to beforeAll and beforeEach methods.

Reply:
done in latest commit